composer/loggers/mosaicml_logger.py

```diff
@@ -96,10 +115,63 @@ def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
         self._log_metadata(metrics)
 
+    def log_analytics(self, state: State) -> None:
+        if self.analytics_data is None:
+            return
+
+        metrics: Dict[str, Any] = {
+            'composer/autoresume': self.analytics_data.autoresume,
+            'composer/precision': state.precision,
+        }
+        metrics['composer/eval_loaders'] = []
+        for evaluator in state.evaluators:
+            dataloader = evaluator.dataloader.dataloader
+            if isinstance(dataloader, torch.utils.data.DataLoader):
+                metrics['composer/eval_loaders'].append(evaluator.label)
+
+        metrics['composer/optimizers'] = [{
+            optimizer.__class__.__name__: optimizer.defaults,
+        } for optimizer in state.optimizers]
+        metrics['composer/algorithms'] = [algorithm.__class__.__name__ for algorithm in state.algorithms]
+
+        metrics['composer/loggers'] = [logger.__class__.__name__ for logger in self.analytics_data.loggers]
+
+        # Take the service provider out of the URI and log it to metadata. If no service provider
+        # is found (i.e. backend = ''), then we assume 'local' for the cloud provider.
+        if self.analytics_data.load_path is not None:
+            backend, _, _ = parse_uri(self.analytics_data.load_path)
+            metrics['composer/cloud_provided_load_path'] = backend if backend else 'local'
+        if self.analytics_data.save_folder is not None:
+            backend, _, _ = parse_uri(self.analytics_data.save_folder)
+            metrics['composer/cloud_provided_save_folder'] = backend if backend else 'local'
+
+        # Save interval can be passed in w/ multiple types. If the type is a function, then
+        # we log 'callable' as the save_interval value for analytics.
+        if isinstance(self.analytics_data.save_interval, Union[str, int]):
+            save_interval_str = str(self.analytics_data.save_interval)
+        elif isinstance(self.analytics_data.save_interval, Time):
+            save_interval_str = f'{self.analytics_data.save_interval._value}{self.analytics_data.save_interval._unit}'
+        else:
+            save_interval_str = 'callable'
+        metrics['composer/save_interval'] = save_interval_str
+
+        if state.fsdp_config:
+            # Keys need to be sorted so they can be parsed consistently in SQL queries
+            metrics['composer/fsdp_config'] = json.dumps(state.fsdp_config, sort_keys=True)
+
+        if state.fsdp_state_dict_type is not None:
+            metrics['composer/state_dict_type'] = state.fsdp_state_dict_type
+
+        self.log_metrics(metrics)
+        self._flush_metadata(force_flush=True)
+
     def log_exception(self, exception: Exception):
         self._log_metadata({'exception': exception_to_json_serializable_dict(exception)})
         self._flush_metadata(force_flush=True)
 
+    def init(self, state: State, logger: Logger) -> None:
+        self.log_analytics(state)
```

Comment:
can we try/except this? I don't want to crash because we failed to log analytics data